contracts/strategies/float/FloatStrategyETH.sol

```diff
@@ -0,0 +1,196 @@
+
+  address[] public liquidationPath;
+
+  // This is only used in `investAllUnderlying()`
+  // The user can still freely withdraw from the strategy
+  modifier onlyNotPausedInvesting() {
+    require(!pausedInvesting, "Action blocked as the strategy is in emergency state");
+    _;
+  }
+
+  constructor(
+    address _storage,
+    address _underlying,
+    address _vault,
+    address _rewardToken,
+    address _uniswapRouterAddress
+  )
+  StrategyBase(_storage, _underlying, _vault, _rewardToken, _uniswapRouterAddress)
+  public {
+    rewardToken = _rewardToken;
+    _weth = address(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+  }
+
+  function depositArbCheck() public view returns(bool) {
+    return true;
+  }
+
+  /*
+  *   In case there are some issues discovered about the pool or underlying asset
+  *   Governance can exit the pool properly
+  *   The function is only used for emergency to exit the pool
+  */
+  function emergencyExit() public onlyGovernance nonReentrant {
+    rewardPool.exit();
+    WETH9 weth = WETH9(address(_weth));
+    weth.deposit.value(address(this).balance)();
+    pausedInvesting = true;
+  }
+
+  /*
+  *   Resumes the ability to invest into the underlying reward pools
+  */
+
+  function continueInvesting() public onlyGovernance {
+    pausedInvesting = false;
+  }
+
+  /**
+  * Sets the route for liquidating the reward token to the underlying token
+  */
+  function setLiquidationPath(address[] memory _newPath) public onlyGovernance {
+    liquidationPath = _newPath;
+  }
+
+  // We assume that all the tradings can be done on Uniswap
+  function _liquidateReward() internal {
+    uint256 rewardAmount = IERC20(rewardToken).balanceOf(address(this));
+
+    if (rewardAmount > 0 // we have tokens to swap
+      && liquidationPath.length > 1 // and we have a route to do the swap
+    ) {
+      notifyProfitInRewardToken(rewardAmount);
+      rewardAmount = IERC20(rewardToken).balanceOf(address(this));
+
+      // we can accept 1 as minimum because this is called only by a trusted role
+      uint256 amountOutMin = 1;
+
+      IERC20(rewardToken).safeApprove(uniswapRouterV2, 0);
+      IERC20(rewardToken).safeApprove(uniswapRouterV2, rewardAmount);
+
+      IUniswapV2Router02(uniswapRouterV2).swapExactTokensForTokens(
+        rewardAmount,
+        amountOutMin,
+        liquidationPath,
+        address(this),
+        block.timestamp
+      );
+    }
+  }
+
+  /*
+  *   Stakes everything the strategy holds into the reward pool
+  */
+  function investAllUnderlying() internal onlyNotPausedInvesting nonReentrant {
+    // this check is needed, because most of the SNX reward pools will revert if
+    // you try to stake(0).
+
+    uint256 underlyingBalance = IERC20(underlying).balanceOf(address(this));
+    if(underlyingBalance > 0) {
+      WETH9 weth = WETH9(address(_weth));
+      weth.withdraw(underlyingBalance); // Unwrapping
+      rewardPool.stake.value(underlyingBalance)(underlyingBalance);
```

Comment:
(underlyingBalance)(underlyingBalance);
why twice?

Reply:
The first one is in the value tag. This adds the given number of ETH to the transaction, so it is needed to send the ETH. The second is the function input for the `stake` function that is called. It is just a number as input. The function in the rewardPool will check if the numbers are the same and everything is okay.